alicloud/7.0/ha/main.tf

```diff
@@ -0,0 +1,309 @@
+
+data "alicloud_regions" "current_region_ds" {
+  current = true
+}
+
+data "alicloud_zones" "default" {
+}
+
+//Random 3 char string appended to the ened of each name to avoid conflicts
+resource "random_string" "random_name_post" {
+  length           = 3
+  special          = true
+  override_special = ""
+  min_lower        = 3
+}
+
+resource "alicloud_vpc" "vpc" {
+  cidr_block = var.vpc_cidr
+  vpc_name   = "${random_string.random_name_post.result}-vpc"
+}
+
+resource "alicloud_vswitch" "vsw_external" {
+  vpc_id       = alicloud_vpc.vpc.id
+  cidr_block   = var.vswitch_cidr_ext
+  zone_id      = data.alicloud_zones.default.zones[0].id
+  vswitch_name = "terraform-ext"
+}
+
+resource "alicloud_vswitch" "vsw_internal" {
+  vpc_id       = alicloud_vpc.vpc.id
+  cidr_block   = var.vswitch_cidr_int
+  zone_id      = data.alicloud_zones.default.zones[0].id
+  vswitch_name = "terraform-int"
+}
+
+resource "alicloud_vswitch" "vsw_hasync" {
+  vpc_id       = alicloud_vpc.vpc.id
+  cidr_block   = var.vswitch_cidr_hasync
+  zone_id      = data.alicloud_zones.default.zones[0].id
+  vswitch_name = "terraform-sync"
+}
+
+resource "alicloud_vswitch" "vsw_hamgmt" {
+  vpc_id       = alicloud_vpc.vpc.id
+  cidr_block   = var.vswitch_cidr_hamgmt
+  zone_id      = data.alicloud_zones.default.zones[0].id
+  vswitch_name = "terraform-mgmt"
+}
+
+// route table
+resource "alicloud_route_table" "intrt" {
+  depends_on       = [alicloud_vswitch.vsw_internal]
+  vpc_id           = alicloud_vpc.vpc.id
+  route_table_name = "int-rt"
+  description      = "internal route table"
+}
+
+resource "alicloud_route_table_attachment" "rtattach" {
+  depends_on     = [alicloud_route_table.intrt]
+  vswitch_id     = alicloud_vswitch.vsw_internal.id
+  route_table_id = alicloud_route_table.intrt.id
+}
+
+// Egress Route to Fortigate
+resource "alicloud_route_entry" "egress" {
+  depends_on = [alicloud_route_table.intrt]
+  // The Default Route
+  route_table_id = alicloud_route_table.intrt.id
+  //route_table_id        = alicloud_vpc.vpc.route_table_id
```

Comment:
do you still need this?

Reply:
yeah, still do.  Since it's a-p, it will failover the route table.